packages/application-shell/src/utils/http-client.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * This file contains helper functions to configure an HTTP client
+ * with the recommended configuration for the Merchant Center API.
+ */
+
+import omitEmpty from 'omit-empty-es';
+import createHttpUserAgent from '@commercetools/http-user-agent';
+import {
+  type ApplicationWindow,
+  STATUS_CODES,
+} from '@commercetools-frontend/constants';
+import { SUPPORTED_HEADERS } from '../constants';
+import * as oidcStorage from './oidc-storage';
+import getCorrelationId from './get-correlation-id';
+import getMcApiUrl from './get-mc-api-url';
+import selectProjectKeyFromUrl from './select-project-key-from-url';
+import selectUserId from './select-user-id';
+
+declare let window: ApplicationWindow;
+
+export type THeaders = Record<string, string>;
+
+export type TConfig = {
+  /**
+   * A custom user agent to identify the HTTP client.
+   * We recommend to use the `@commercetools/http-user-agent` package.
+   *
+   * @example
+   *
+   * import createHttpUserAgent from '@commercetools/http-user-agent';
+   *
+   * const userAgent = createHttpUserAgent({
+   *   name: 'fetch-client',
+   *   version: '2.6.0',
+   *   libraryName: window.app.applicationName,
+   *   contactEmail: 'support@my-company.com',
+   * });
+   */
+  userAgent?: string;
+
+  /**
+   * Additional headers to be included in the request.
+   * The provided recommended headers won't be overwritten.
+   * See `TOptions.headers`.
+   */
+  headers?: THeaders;
+};
+
+export type TOptions = {
+  credentials: 'include';
+
+  /**
+   * The HTTP headers included by default are:
+   * - Accept
+   * - Authorization (only in development)
+   * - X-Application-Id
+   * - X-Correlation-Id
+   * - X-Project-Key
+   * - X-User-Agent
+   */
+  headers: THeaders;
+};
+
+export type TFetcherResponse<Data> = {
+  /**
+   * The parsed response from the server.
+   */
+  data: Data;
+  /**
+   * The HTTP status code from the server response.
+   */
+  statusCode: number;
+  /**
+   * Implement a function to access the HTTP headers from the server response.
+   */
+  getHeader: (headerName: string) => string | null;
```

Comment:
nitpick: I was thinking maybe this property should be optional. The reason is that maybe users don't need to access response headers in their use cases and this way we would be asking them to always implement this function.

Reply:
This property should be required because is needed for this:

https://github.com/commercetools/merchant-center-application-kit/pull/2734/files#diff-2ea8fac16a771a4733de24f7b316e01175f44514f2baf4d4715f2443cfbef1c4R146-R149

The main point here is that HTTP clients might return different type of response objects and headers, so we can't reliably access the header.

Or we can try to "guess" but the user still needs to explicitly return the `headers` to the function.

Either way, something must be returned.

Reply:
Oh! I see.
Thanks for the explanation 👍